src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.ts

```diff
@@ -204,6 +207,7 @@ export const useRespondentAnswers = () => {
 
         setValue('flowSubmissions', flowSubmissions);
         setValue('flowResponses', flowResponses);
+        setValue('flowResponseOptionsCount', responseOptionsCount);
```

Comment:
As I see this is an incorrect use of the `react-hook-form`, not for its intended purpose: for each new field the library will create the bunch of attributes and handlers. That means each time user change the real input it will do the redundant checks with such fields and will side effect the real state of the form.

As I see we can use own context here.

Reply:
Perfect!